### DISCUSSION TITLE
How to diagnose and improve performances of Automations

### DISCUSSION THREAD

**Discussion by Alan-Guggenheim** [zwave-js/zwave-js#4246] [Q&A]
I have 81 Z-Wave devices (454 Entities) many are battery operated motion or door sensors and are used to turn lights on and off.
Several issues I am chasing:
1- After an update or any reboot, devices do not respond. I need for example to open the door, close and re-open for the light to turn on.
2- If there is any pending update, the automations are very slow ( 5 to 30 seconds or more). Once HA updated, they go back to normal
3- Some rooms are stubborn and always take too long to turn on.
Any ideas of troubleshooting or fixes?
Thanks

**Comment by AlCalzone** [comment]
1 and 3 needs driver logs to diagnose anything. I suggest you read https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index

2 feels coincidental. The z-wave driver is running in a separate process from HA, so if the update has anything to do with it, there must be something in HA itself going on.

For 3, you could also check out the new health check tools in zwavejs2mqtt (in the network map)
